app/Models/Signup.php

```diff
@@ -248,4 +248,25 @@ public function scopeWithVisiblePosts($query, $types = null)
             }
         }]);
     }
+
+    /**
+     * Gets event payload for a referral signup, on behalf of the referrer user ID.
+     */
+    public function getReferralSignupEventPayload()
+    {
+        $userId = $this->northstar_id;
+        $user = app(GraphQL::class)->getUserById($userId);
+
+        $campaignWebsite = app(GraphQL::class)->getCampaignWebsiteByCampaignId($this->campaign_id);
+
+        return [
+            'id' => $this->id,
+            'user_id' => $userId,
+            'user_display_name' => isset($user) ? $user['displayName'] : null,
+            'campaign_id' => (string) $this->campaign_id,
```

Comment:
No biggie just curious to why is this being cast as a string?

Reply:
I was copying the format from the main Blink event for signups: https://github.com/DoSomething/rogue/blob/8627ed64be6f88d035c556be34e5d913485aa015/app/Models/Signup.php#L201

Buuuut, I _believe_ that's a Blink concern, and this is going directly to Customer.io so I wonder if the rationale is moot?

Reply:
🤔 Though if c.io is expecting a string, we might need to remain consistent in any case?

Reply:
Yeah, makes sense to leave since that's what we're sending for the `campaign_signup` events.